code/modules/reagents/reagent_containers/food/snacks/egg.dm

```diff
@@ -21,11 +21,15 @@
 
 /obj/item/weapon/reagent_containers/food/snacks/egg/throw_impact(atom/hit_atom)
 	..()
-	new/obj/effect/decal/cleanable/egg_smudge(src.loc)
-	src.reagents.reaction(hit_atom, TOUCH)
-	src.visible_message("<span class='warning'>\The [src.name] has been squashed.</span>","<span class='warning'>You hear a smack.</span>")
-	playsound(src.loc, 'sound/items/egg_squash.ogg', 50, 1)
-	qdel(src)
+	if(isturf(hit_atom))
```

Comment:
I love throwcode.

1. `throw_impact()` is called for EVERY ATOM on the tile on which the item stops
2. It includes one of the at least three (probably four, maybe even more) separate implementations of the Bresenham line algorithm present in our code when we only really need two at most (one to get a list of turfs in the line, one like the `bresenham_step()` projectiles have except not as completely and utterly incomprehensible)
3. It magically just werks with projectiles in some ways but catastrophically fails in others (mostly due to point 1)
4. This isn't *directly* related to thrown item code, but the way armor protects you from thrown items is monumentally stupid. (You have a chance to take no damage equal to the square of the armor's melee protection. Otherwise you take full damage.)